packages/react-router/docs/blocked-updates.md

```diff
@@ -0,0 +1,97 @@
+# Dealing with Update Blocking
+
+React's `shouldComponentUpdate` lifecycle method optimizes rendering performance by blocking the re-rendering of components. Unfortunately, this means that React Router's location-aware components can become out of sync with the current location if their re-rendering was prevented. For example, a `<NavLink>` that is "active" for a location may continue to be styled as active after a location change, despite no longer matching the current URL, because it is not updated when the location changed.
+
+### Example of the Problem
+
+When the `<UpdateBlocker>` is mounting, any location-aware child components will use the current location and match information to render.
+
+```js
+// location = { pathname: '/about' }
+<UpdateBlocker>
+  <NavLink to='/about' activeClassName='active'>About</NavLink>
+  // <a href='/about' class='active'>About</a>
+  <NavLink to='/faq' activeClassName='active'>F.A.Q.</NavLink>
+  // <a href='/faq'>F.A.Q.</a>
+</UpdateBlocker>
+```
+
+When the location changes, the `<UpdateBlocker>`'s `shouldComponentUpdate` method will return `false`, and any children inside of it will not be re-rendered
+
+```js
+// location = { pathname: '/faq' }
+<UpdateBlocker>
+  // the links will not re-render, so they retain their previous attributes
+  <NavLink to='/about' activeClassName='active'>About</NavLink>
+  // <a href='/about' class='active'>About</a>
+  <NavLink to='/faq' activeClassName='active'>F.A.Q.</NavLink>
+  // <a href='/faq'>F.A.Q.</a>
+</UpdateBlocker>
+```
+
+### Third-Party Code
+
+You may run into issues of components not updating after a location change despite not calling `shouldComponentUpdate` yourself. This is most likely because `shouldComponentUpdate` is being called by third-party code, such as `react-redux`'s `connect` and `mobx-react`'s `observer`.
```

Comment:
I'm going to try and get on a 5.1 of react-redux this week that gets rid of sCU, so hopefully you can remove this section ;)